### PR TITLE
Create site tags

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add site tag model
- Make tag unique for site tag
- Remove blank tags and treat empty tags as nil
- Show site tags minus disabled track tags as suggestions

### CODE REVIEW

File: app/commands/training_data/code_tags_sample/normalize_tags.rb [hunk 1]
```diff
@@ -5,8 +5,13 @@ class TrainingData::CodeTagsSample::NormalizeTags
 
   def call
     return nil if tags.nil?
+    return nil if normalized_tags.empty?
```
Comment (ErikSchierboom) [inline app/commands/training_data/code_tags_sample/normalize_tags.rb:8]: I felt that treating empty tags as `nil` makes things a bit easier (e.g. querying)

File: app/commands/training_data/code_tags_sample/normalize_tags.rb [hunk 2]
```diff
@@ -5,8 +5,13 @@ class TrainingData::CodeTagsSample::NormalizeTags
...
 
-    tags.flat_map { |tag| MAPPING.fetch(tag, tag) }.uniq
+    normalized_tags
+  end
+
+  def normalized_tags
+    tags.flat_map { |tag| MAPPING.fetch(tag, tag) }.reject(&:blank?).uniq
```
Comment (ErikSchierboom) [inline app/commands/training_data/code_tags_sample/normalize_tags.rb:14]: I'm now also removing blank tags

File: db/migrate/20231201091435_create_site_tags.rb [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    return if Rails.env.production?
+
+    create_table :site_tags do |t|
+      t.string :tag, null: false, index: { unique: true }
+      t.string :description, null: false
+
+      t.timestamps
+    end
+
+    [
```
Comment (ErikSchierboom) [inline db/migrate/20231201091435_create_site_tags.rb:12]: Let me know if you don't want this in the migration. It is an easy way for us to get the data locally 🤷

Comment (iHiD) [reply db/migrate/20231201091435_create_site_tags.rb:12]: It's fine.

**Pull request merged**